src/Networking/NexusMods.Networking.ModUpdates/PerFeedCacheUpdater.cs

```diff
@@ -0,0 +1,155 @@
+using System.Diagnostics;
+using NexusMods.Abstractions.NexusWebApi.Types.V2;
+using NexusMods.Networking.ModUpdates.Private;
+using NexusMods.Networking.ModUpdates.Traits;
+namespace NexusMods.Networking.ModUpdates;
+
+/// <summary>
+/// This is a helper struct which helps us make use of the 'most recently updated mods for game'
+/// API endpoint to use as a cache. (Where 'game' is a 'feed')
+/// For full info on internals, see the NexusMods.App project documentation.
+///
+/// This API consists of the following:
+///
+/// 1. Input [Constructor]: A set of items with a 'last update time' (see <see cref="ICanGetLastUpdatedTimestamp"/>)
+///           and a 'unique id' (see <see cref="ICanGetUidForMod"/>) that are relevant to the current 'feed' (game).
+/// 
+/// 2. Update [Method]: Submit results from API endpoint returning 'most recently updated mods for game'.
+///           This updates the internal state of the <see cref="MultiFeedCacheUpdater{TUpdateableItem}"/>.
+/// 
+/// 3. Output [Info]: The <see cref="MultiFeedCacheUpdater{TUpdateableItem}"/> outputs items with 2 categories:
+///        - Up-to-date mods. These should have their timestamp updated.
+///        - Out of date mods. These require re-querying the data from external source.
+/// </summary>
+/// <remarks>
+/// Within the Nexus Mods App:
+///
+/// - 'Input' is our set of locally cached mod pages.
+/// - 'Update' is our results of `updated.json` for a given game domain.
+/// - 'Output' are the pages we need to update.
+///
+/// The 'Feed' in the context of the Nexus App is the individual game's 'updated.json' endpoint;
+/// i.e. a 'Game Mod Feed'
+/// </remarks>
+public class PerFeedCacheUpdater<TUpdateableItem> where TUpdateableItem : ICanGetLastUpdatedTimestamp, ICanGetUidForMod
```

Comment:
This code project is overkill for the task at hand. Instead implement what we can on the V2 API and put that code inside the NexusWebApi project.

When we over-abstract like this we endup with a lot of code that has to be assembled to do something that is only ever used with one set of data. We have a "Per feed updater" that can update any number of things, mixins for Mod and Page metadata, `ICanGet...` interfaces. All that can be reduced to a few small method in the API class if we assume that what we're caching is mod info from the V2 API